### PR TITLE
feat: add clean-toolchain-resources-in-dev and move clean targets to one file

### PR DESCRIPTION
**Pull request by MatousJobanek**
This PR:
* adds  clean-toolchain-resources-in-dev target deleting the toolchain resources
* moves all clean* targets to clean.mk file

### CODE REVIEW

File: make/clean.mk [hunk 1]
```diff
@@ -1,4 +1,42 @@
...
+	$(Q)for CRD in `oc get crd -o name -n $(DEV_HOST_NS) | grep toolchain`; do \
+		echo "deleting $${CRD} in $(DEV_HOST_NS)"; \
+		CRD_NAME=`oc get $${CRD} --template '{{.metadata.name}}'`; \
+		oc get $${CRD_NAME} -n $(DEV_HOST_NS) -o name | xargs -i oc delete {}; \
+	done
+	$(Q)for CRD in `oc get crd -o name -n $(DEV_MEMBER_NS) | grep toolchain`; do \
+		echo "deleting $${CRD} in $(DEV_MEMBER_NS)"; \
+		CRD_NAME=`oc get $${CRD} --template '{{.metadata.name}}'`; \
+		oc get $${CRD_NAME} -n $(DEV_MEMBER_NS) -o name | xargs -i oc delete {}; \
+	done
```
Comment (alexeykazakov) [inline make/clean.mk:39]:
```suggestion
	$(Q)for CRD in `oc get crd -o name | grep toolchain`; do \
		echo "deleting $${CRD} in $(DEV_HOST_NS) and $(DEV_MEMBER_NS)"; \
		CRD_NAME=`oc get $${CRD} --template '{{.metadata.name}}'`; \
		oc get $${CRD_NAME} -n $(DEV_HOST_NS) -o name | xargs -i oc delete {}; \
		oc get $${CRD_NAME} -n $(DEV_MEMBER_NS) -o name | xargs -i oc delete {}; \
	done
```

File: make/clean.mk [hunk 2]
```diff
@@ -1,4 +1,42 @@
...
+.PHONY: clean-e2e-files
+## Removes files and directories used during e2e test setup
+clean-e2e-files:
+	rm -f ${WAS_ALREADY_PAIRED_FILE} 2>/dev/null || true
+	rm -rf ${IMAGE_NAMES_DIR} 2>/dev/null || true
+
+.PHONY: clean-toolchain-resources-in-dev
+## Deletes resources in the OpenShift cluster. The deleted resources are:
+##    * namesapces created only during the dev setup (for both operators host and member)
+##    * all toolchain.dev.openshift.com CRs in both host a member namespaces
```
Comment (alexeykazakov) [inline make/clean.mk:24]: It doesn't delete the namespaces itself.
```suggestion
##    * all toolchain.dev.openshift.com CRs in both host and member namespaces created during the dev setup
```

File: make/clean.mk (hunk 1 above)
Comment (MatousJobanek) [reply make/clean.mk:39]: good point :man_facepalming: :sweat_smile:

File: make/clean.mk [hunk 3]
```diff
@@ -1,4 +1,37 @@
...
+## Removes vendor directory and runs go clean command
 clean:
 	$(Q)-rm -rf ${V_FLAG} $(OUT_DIR) ./vendor
-	$(Q)go clean ${X_FLAG} ./...
\ No newline at end of file
+	$(Q)go clean ${X_FLAG} ./...
+
+.PHONY: clean-e2e-resources
+## Deletes resources in the OpenShift cluster. The deleted resources are:
+##    * namesapces created during both the dev and e2e test setup (for both operators host and member)
```
Comment (xcoulon) [inline make/clean.mk:9]:
```suggestion
##    * namespaces created during both the dev and e2e test setup (for both operators host and member)
```

File: make/clean.mk [hunk 4]
```diff
@@ -1,4 +1,37 @@
...
+
+.PHONY: clean-toolchain-resources-in-dev
+## Deletes resources in the OpenShift cluster. The deleted resources are:
+##    * all toolchain.dev.openshift.com CRs in both host and member namespaces created during the dev setup
+##    * all ClusterresourceQuotas with the label toolchain.dev.openshift.com/provider=codeready-toolchain in member namespace
+##    * pods of both operators host and member
+clean-toolchain-resources-in-dev:
+	$(Q)echo "cleaning resources in $(DEV_HOST_NS) and $(DEV_MEMBER_NS)..."
+	$(Q)oc get usersignups -n $(DEV_HOST_NS) -o name | xargs -i oc delete -n $(DEV_HOST_NS) {}
+	$(Q)for CRD in `oc get crd -o name | grep toolchain`; do \
```
Comment (xcoulon) [inline make/clean.mk:29]: Ideally, removing the `usersignup` resources should be enough, but I guess it's fine if we delete everything explicitly (and our operator controller should be able to handle such deletions)

Comment (MatousJobanek) [reply make/clean.mk:29]: removing the `usersignups` won't delete some resources such as `BannedUser`, `NSTemplateTier` etc... And the next lines that delete the operator pods will trigger the next "fresh" deployment so the Tiers should be created again

**Pull request merged**